packages/opentelemetry-core/src/context/propagation/B3Propagator.ts

```diff
@@ -71,29 +94,38 @@ export class B3Propagator implements HttpTextPropagator {
   extract(context: Context, carrier: unknown, getter: GetterFunction): Context {
     const traceIdHeader = getter(carrier, X_B3_TRACE_ID);
     const spanIdHeader = getter(carrier, X_B3_SPAN_ID);
+    const parentSpanIdHeader = getter(carrier, X_B3_PARENT_SPAN_ID);
     const sampledHeader = getter(carrier, X_B3_SAMPLED);
+    const flagsHeader = getter(carrier, X_B3_FLAGS);
 
-    const traceIdHeaderValue = Array.isArray(traceIdHeader)
-      ? traceIdHeader[0]
-      : traceIdHeader;
-    const spanId = Array.isArray(spanIdHeader) ? spanIdHeader[0] : spanIdHeader;
+    const traceIdHeaderValue = parseHeader(traceIdHeader);
+    const spanId = parseHeader(spanIdHeader);
+    const parentSpanId = parseHeader(parentSpanIdHeader);
+    const options = parseHeader(sampledHeader);
+    const debugHeaderValue = parseHeader(flagsHeader);
+    const debug = debugHeaderValue === '1';
+    const traceFlagsOrDebug = Number(debug) || Number(options);
```

Comment:
Per https://github.com/openzipkin/b3-propagation#sampling-state-1, some tracers propagated `X-B3-Sampled` as `true` or `false` as opposed to `1` or `0`. I think we should accept these["True", "False", "1", "0"] values.

Reply:
This already happens

Reply:
There's this test 'should extract context of a sampled span from carrier when sampled is mentioned as boolean true flag'

Reply:
hmm, but I'm guessing maybe that test isn't working as expected because it would be a string ad not a boolean in the carrier

Reply:
Yeah, I think it would actually be a string when the carrier comes in, so we also need tests for that.